src/pool.jl

```diff
@@ -62,6 +67,9 @@ avoid doing a dict lookup twice
 
     i = R(n + 1)
     push!(pool.levels, x)
+    pool.hash = hash(x, pool.hash)
+    pool.equalto = 0
```

Comment:
are we guarantteed that `objectid` does not return `0`?

Reply:
Probably not. The safe approach would be to use `nothing` as a sentinel. But actually `objectid` merely hashes the pointer to the pool, so it would be simpler to just store the pointer. That should even be faster since we call this repeatedly in hot loops.